sgkit/stats/aggregation.py

```diff
@@ -477,9 +502,187 @@ def count_variant_genotypes(
     new_ds = create_dataset(
         {variables.variant_genotype_count: (("variants", "genotypes"), C)}
     )
+    if assign_coords is True and merge is False:
```

Comment:
Could be simplified to `if assign_coords and not merge`.

What happens if `merge` is true? Are the coordinates assigned in that case?

Reply:
Fixed.

Yes, the coords will already exist in `ds` from the initial `define_variable_if_absent` of `genotype_coords`. Any coords in `ds` are kept when it's merged with `new_ds` (checked in `test_genotype_coords`).